main/maven/src/mill/main/maven/Plugins.scala

```diff
@@ -0,0 +1,126 @@
+package mill.main.maven
+
+import org.apache.maven.model.{Model, Plugin, PluginExecution}
+import org.codehaus.plexus.util.xml.Xpp3Dom
+
+import scala.jdk.CollectionConverters.*
+
+/** Utilities for handling Maven plugins. */
+@mill.api.internal
+object Plugins {
+
+  def find(model: Model, groupId: String, artifactId: String): Option[Plugin] =
+    model.getBuild.getPlugins.asScala
+      .find(p => p.getGroupId == groupId && p.getArtifactId == artifactId)
+
+  def dom(plugin: Plugin): Option[Xpp3Dom] =
+    plugin.getConfiguration match {
+      case xpp3: Xpp3Dom => Some(xpp3)
+      case _ => None
+    }
+
+  def dom(execution: PluginExecution): Option[Xpp3Dom] =
+    execution.getConfiguration match {
+      case xpp3: Xpp3Dom => Some(xpp3)
+      case _ => None
+    }
+
+  /**
+   * @see [[https://maven.apache.org/plugins/maven-compiler-plugin/index.html]]
+   */
+  object `maven-compiler-plugin` {
+
```

Comment:
Let's give these `camelCase` names; I think the correspondence with the `kebab-case` maven plugin name is clear enough, especially with a comment linking to the relevant website